### PR TITLE
Update DG Architecture portion

### PR DESCRIPTION
**Pull request by yiwen101**
Modify the architecture section of DG.

Remove Main and common from diagram (change to other notes part in DG as they are not important) to help readers to focus

Add explanation on the duties of each component and the relationship between them.
Better organize the contents in the architecture section into sub sections

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -29,42 +29,71 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 <puml src="diagrams/ArchitectureDiagram.puml" width="280" />
 
-The ***Architecture Diagram*** given above explains the high-level design of the App.
+The ***Architecture Diagram*** given above illustrates the high-level design of the App. 
 
-Given below is a quick overview of main components and how they interact with each other.
+You could see that The bulk of the app's works are done by the UI, Logic, Model and Storage components. Below are the general duties of each component and how they interact with each other.
```
Comment (richiehx) [inline docs/DeveloperGuide.md]: Some grammar issues: could -> can, The -> the

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -29,42 +29,71 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
...
 
-**Main components of the architecture**
+#### Duties of each component
+[**`UI`**](#ui-component) is responsible for
+* Rendering visuals to the user.
+* Handling and responding to user interactions.
 
-**`Main`** (consisting of classes [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/MainApp.java)) is in charge of the app launch and shut down.
```
Comment (richiehx) [inline docs/DeveloperGuide.md:38]: I think we should still include this as it is the component that is running the application.

**Review by richiehx — COMMENTED** [review]
Some questions regarding the removal of the Main class from the diagram.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (pzl111) [inline docs/DeveloperGuide.md]: i think it should be "the" not "The"

**Review by pzl111 — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (pzl111) [reply docs/DeveloperGuide.md:38]: its included below

**Pull request merged**